frontend/src/components/datasets/PublicDataset.tsx

```diff
@@ -339,7 +358,13 @@ export const PublicDataset = (): JSX.Element => {
 						) : (
 							<></>
 						)}
-						<FilesTable datasetId={datasetId} folderId={folderId} publicView={true} />
+						<FilesTable
+							datasetId={datasetId}
+							folderId={folderId}
+							filesInDataset={filesInDataset}
+							foldersInDataset={foldersInDataset}
+							publicView={true}
```

Comment:
I modified the FilesTable component passing the files and folder as props instead for the pagination purpose. Hence modified